lang/cs/Org.Apache.REEF.Wake/Remote/Impl/WritableRemoteManager.cs

```diff
@@ -0,0 +1,318 @@
+﻿/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using Org.Apache.REEF.Utilities.Logging;
+using Org.Apache.REEF.Wake.Util;
+
+namespace Org.Apache.REEF.Wake.Remote.Impl
+{
+    /// <summary>
+    /// Manages incoming and outgoing messages between remote hosts.
+    /// </summary>
+    /// <typeparam name="T">Message type T. It is assumed to be IWritable</typeparam>
+    [Obsolete("Need to remove Iwritable and use IstreamingCodec. Please see Jira REEF-295 ", false)]
+    public sealed class WritableRemoteManager<T> : IRemoteManager<T> where T : IWritable
+    {
+        private static readonly Logger LOGGER = Logger.GetLogger(typeof (DefaultRemoteManager<T>));
+
+        private readonly WritableObserverContainer<T> _observerContainer;
+        private readonly WritableTransportServer<IWritableRemoteEvent<T>> _server;
+        private readonly Dictionary<IPEndPoint, ProxyObserver> _cachedClients;
+
+        /// <summary>
+        /// Constructs a DefaultRemoteManager listening on the specified address and any
+        /// available port.
+        /// </summary>
+        /// <param name="localAddress">The address to listen on</param>
+        [Obsolete("Use IRemoteManagerFactory.GetWritableInstance() instead.", false)]
+        public WritableRemoteManager(IPAddress localAddress) : this(localAddress, 0)
```

Comment:
Given that this is new code, you only need the constructors offered through the Factory. You can pair these down to the one most specific one, and put all the default logic into the Factory. Also, make that one constructor `internal`, as it shall only be called by the Factory.